api/src/pcapi/routes/native/v1/account.py

```diff
@@ -373,6 +374,22 @@ def suspend_account(user: users_models.User) -> None:
     transactional_mails.send_user_request_to_delete_account_reception_email(user)
 
 
+@blueprint.native_v1.route("/account/suspend_for_suspicious_login", methods=["POST"])
+@spectree_serialize(api=blueprint.api, on_success_status=204, on_error_statuses=[400, 401, 404])
+def suspend_account_for_suspicious_login(body: serializers.SuspendAccountForSuspiciousLoginRequest) -> None:
+    try:
+        decoded_token = decode_jwt_token(body.token)
+        user = users_models.User.query.filter_by(id=decoded_token["userId"]).one()
+        api.suspend_account(user, constants.SuspensionReason.SUSPICIOUS_LOGIN_REPORTED_BY_USER, actor=user)
+        transactional_mails.send_user_request_to_delete_account_reception_email(user)
+    except jwt.ExpiredSignatureError:
+        raise api_errors.ApiErrors({"reason": "Le token a expiré."}, status_code=401)
+    except (jwt.InvalidTokenError, jwt.InvalidSignatureError):
+        raise api_errors.ApiErrors({"reason": "Le token est invalide."})
+    except sa.orm.exc.NoResultFound:
+        raise api_errors.ResourceNotFoundError()
```

Comment:
(détail / style) On préfère généralement n'entourer avec des try/except que les lignes concernées par chaque expression. Ça évite d'attraper une erreur en pensant qu'elle est levée par la ligne X et d'avoir un comportement particulier, alors que l'erreur est levée par la ligne Y, pour laquelle on devrait avoir un autre comportement. Ici, ça devrait donner quelque chose comme ça : 

```suggestion
    try:
        decoded_token = decode_jwt_token(body.token)
    except jwt.ExpiredSignatureError:
        raise api_errors.ApiErrors({"reason": "Le token a expiré."}, status_code=401)
    except (jwt.InvalidTokenError, jwt.InvalidSignatureError):
        raise api_errors.ApiErrors({"reason": "Le token est invalide."})
    try:
        user = users_models.User.query.filter_by(id=decoded_token["userId"]).one()
    except sa.orm.exc.NoResultFound:
        raise api_errors.ResourceNotFoundError()
    api.suspend_account(user, constants.SuspensionReason.SUSPICIOUS_LOGIN_REPORTED_BY_USER, actor=user)
    transactional_mails.send_user_request_to_delete_account_reception_email(user)
```

Reply:
Top ! Effectivement c'est plus facile à lire comme ça